src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -704,6 +784,34 @@ impl<S: StateStore, SD: ValueRowSerde> StorageTableInner<S, SD> {
 
         Ok(iter)
     }
+
+    /// Iterates on the table with the given prefix of the pk in `pk_prefix` and the range bounds.
+    /// Returns a stream of `DataChunk` with the provided `chunk_size`
+    pub async fn batch_chunk_iter_with_pk_bounds(
```

Comment:
IIUC, the current implementation is row-store oriented. If a column store is supported in the future, it can use this interface to pass the data chunk to the batch executor more efficiently.